services/repository/transfer.go

```diff
@@ -44,7 +44,7 @@ func TransferOwnership(doer, newOwner *user_model.User, repo *repo_model.Reposit
 	}
 	repoWorkingPool.CheckOut(fmt.Sprint(repo.ID))
 
-	newRepo, err := repo_model.GetRepositoryByID(repo.ID)
+	newRepo, err := repo_model.GetRepositoryByID(db.DefaultContext, repo.ID)
```

Comment:
Or what kinds of functions need the context especially?
Because you seem to have it added mainly for `GET` functions, and not for `UPDATE/ DELETE` functions.
So, is there a guideline what should get a ctx parameter, or is it simply whatever we think needs it?

Reply:
> Or what kinds of functions need the context especially? Because you seem to have it added mainly for `GET` functions, and not for `UPDATE/ DELETE` functions. So, is there a guideline what should get a ctx parameter, or is it simply whatever we think needs it?

This need a separate refactor PR, we need move some codes from `models.TransferOwnership` to service layer `TransferOwnership`.